src/gasPriceOracle/oracle.ts

```diff
@@ -1,51 +1,93 @@
+  transport?: Transport;
+}
+
+interface EthersGasPriceEstimateOptions extends GasPriceEstimateOptions {
+  chainId: number;
+}
+
+interface ViemGasPriceEstimateOptions extends Partial<GasPriceEstimateOptions> {
+  baseFeeMultiplier: number;
+}
+
+const GAS_PRICE_ESTIMATE_DEFAULTS: GasPriceEstimateOptions = {
+  baseFeeMultiplier: 1,
+  legacyFallback: true,
+};
 
 /**
  * Provide an estimate for the current gas price for a particular chain.
- * @param chainId The chain ID to query for gas prices.
  * @param provider A valid ethers provider.
- * @param legacyFallback In the case of an unrecognised chain, fall back to type 0 gas estimation.
- * @returns Am object of type GasPriceEstimate.
+ * @param {opts} GasPriceEstimateOptions optional parameters.
+ * @returns An  object of type GasPriceEstimate.
  */
 export async function getGasPriceEstimate(
   provider: providers.Provider,
-  chainId?: number,
-  transport?: Transport,
-  legacyFallback = true
+  opts: Partial<GasPriceEstimateOptions>
 ): Promise<GasPriceEstimate> {
-  chainId ?? ({ chainId } = await provider.getNetwork());
+  const {
+    baseFeeMultiplier,
+    chainId: _chainId,
+    unsignedTx,
+    transport,
+    legacyFallback,
+  }: GasPriceEstimateOptions = {
+    ...GAS_PRICE_ESTIMATE_DEFAULTS,
+    ...opts,
+  };
+  assert(
+    baseFeeMultiplier >= 1.0 && baseFeeMultiplier <= 5,
+    `Require 1.0 < base fee multiplier (${baseFeeMultiplier}) <= 5.0 for a total gas multiplier within [+1.0, +5.0]`
+  );
+
+  const chainId = _chainId ?? (await provider.getNetwork()).chainId;
 
-  const useViem = process.env[`NEW_GAS_PRICE_ORACLE_${chainId}`] === "true";
+  // We only use the unsignedTx in the viem flow.
+  const useViem = chainId === CHAIN_IDs.LINEA && process.env[`NEW_GAS_PRICE_ORACLE_${chainId}`] === "true";
   return useViem
-    ? getViemGasPriceEstimate(chainId, transport)
-    : getEthersGasPriceEstimate(provider, chainId, legacyFallback);
+    ? _getViemGasPriceEstimate(chainId, { baseFeeMultiplier, unsignedTx, transport })
+    : _getEthersGasPriceEstimate(provider, {
+        baseFeeMultiplier,
+        chainId,
+        legacyFallback,
+      });
 }
 
 /**
  * Provide an estimate for the current gas price for a particular chain.
  * @param chainId The chain ID to query for gas prices.
  * @param provider A valid ethers provider.
  * @param legacyFallback In the case of an unrecognised chain, fall back to type 0 gas estimation.
- * @returns Am object of type GasPriceEstimate.
+ * @returns An object of type GasPriceEstimate.
  */
-function getEthersGasPriceEstimate(
+function _getEthersGasPriceEstimate(
   provider: providers.Provider,
-  chainId: number,
-  legacyFallback = true
+  opts: EthersGasPriceEstimateOptions
 ): Promise<GasPriceEstimate> {
+  const { baseFeeMultiplier, chainId, legacyFallback } = opts;
+
   const gasPriceFeeds = {
     [CHAIN_IDs.ALEPH_ZERO]: arbitrum.eip1559,
     [CHAIN_IDs.ARBITRUM]: arbitrum.eip1559,
```

Comment:
@pxrl should Linea still be legacy in the ethers provider case? Currently the linea.eip1559 implementation is just a passthrough to the ethereum.legacy call